pkg/hub/addon/discovery_controller.go

```diff
@@ -127,31 +140,18 @@ func (c *addOnFeatureDiscoveryController) syncAddOn(ctx context.Context, cluster
 		labels[key] = getAddOnLabelValue(addOn)
 	}
 
-	cluster, err := c.clusterLister.Get(clusterName)
-	if errors.IsNotFound(err) {
-		// no cluster, it could be deleted
-		return nil
-	}
+	patchBytes, err := json.Marshal(map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"labels": labels,
+		},
+	})
 	if err != nil {
-		return fmt.Errorf("unable to find cluster with name %q: %w", clusterName, err)
-	}
-	// no work if cluster is deleting
-	if !cluster.DeletionTimestamp.IsZero() {
-		return nil
+		return fmt.Errorf("failed to create patch for cluster %s: %w", cluster.Name, err)
 	}
 
-	// merge labels
-	modified := false
-	cluster = cluster.DeepCopy()
-	resourcemerge.MergeMap(&modified, &cluster.Labels, labels)
-
-	// no work if the cluster labels have no change
-	if !modified {
-		return nil
-	}
```

Comment:
How about keeping the modified check, after the modified check, we create `patchBytes` and apply the patch, thus we can avoid to path the same thing

Reply:
added label equality check, if they are equal, no need to patch.